### ISSUE TITLE
java.lang.OutOfMemoryError

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#308]
I've noticed several of the older phones having this issue:

```
USER_COMMENT=Crashes immediately after opening or after choosing file.
ANDROID_VERSION=4.4.2
APP_VERSION_NAME=1.37
BRAND=samsung
PHONE_MODEL=SM-T311
CUSTOM_DATA=
STACK_TRACE=java.lang.OutOfMemoryError
at com.android.volley.toolbox.ByteArrayPool.getBuf(ByteArrayPool.java:101)
at com.android.volley.toolbox.PoolingByteArrayOutputStream.<init>(PoolingByteArrayOutputStream.java:53)
at com.android.volley.toolbox.BasicNetwork.entityToBytes(BasicNetwork.java:228)
at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:123)
at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:112)
```

Is there anything we can do about it, or it can't be helped?

**Comment by nicolas-raoul** [comment]
What is the usual size of the HTTP request/response coming from this request?

If the crash happens when sending the image, I guess we could avoid loading the full file in memory, and just stream it out while reading it.

**Comment by misaochan** [comment]
> What is the usual size of the HTTP request/response coming from this request?

How do we find this out?

**Comment by nicolas-raoul** [comment]
Either via debugging or with a network inspection tool such as Wireshark.

Does this crash always happen when uploading an image to Commons?

**Comment by misaochan** [comment]
Oh, this doesn't happen for me... I was just observing these crash reports on our forum and noting the Android versions/models tend to be older. I don't know if they would be amenable to helping us test via Wireshark, but I could ask.

**Comment by nicolas-raoul** [comment]
Can we guess at what step that happens, from the stack trace?

**Comment by misaochan** [comment]
I don't think so - the stack trace ends there without any of our classes being mentioned. It appears to be network-related... but pretty much everything in our app falls under that category, heh.

**Comment by whym** [comment]
Probably fixed - see also #94 #607.

**Comment by misaochan** [comment]
Yeah, agree.
